### ISSUE TITLE
LSP: Go to type definition support

### ISSUE THREAD

**Issue by jvilk-stripe** [sorbet/sorbet#1500]
### Problem

Given a variable, `foo`, it can be difficult to jump to the location that defines the *type* of `foo`.

### Proposed solution

[Implement "Go to type definition...".](https://microsoft.github.io/language-server-protocol/specification#textDocument_typeDefinition) The behavior in TypeScript is that it jumps to the location that defines the type of the thing you are pointing at. We should implement the same behavior for Sorbet.

### Open questions

* How should this feature handle complex types? (Union/nilable/etc)
  * It could return the locations for each component of the type...? Or return nothing.
  * TypeScript returns a location for each type in a union, but nothing for intersection types.

### Difference from "go to definition"

Given a variable `foo` of type `Bar`, "Go to definition" goes to the *definition of `foo`*, which is the location where `foo` was last assigned. "Go to type definition" goes to the *definition of `Bar`*, which is the *type* of `foo`.

```ruby
class Bar; end # Definition of foo's type, `Bar`

foo = Bar.new # Definition of foo
```

**Comment by DarkDimius** [comment]
Note that this feature is different from "go to definition"

**Comment by jvilk-stripe** [comment]
(Updated description to clarify)

**Comment by jaredbeck** [comment]
I don't see anything about IDE support in the docs. If I'd like to learn more about IDEs that support sorbet, where could I read more? Thanks.

**Comment by jvilk-stripe** [comment]
@jaredbeck Thanks for your interest! We're still working on making IDE support decent, so we have not documented it yet or made an announcement. When it is released, we will likely make a public announcement. It will work with any editor that supports the language server protocol (either natively or, more typically, via a plugin/extension). Internally at Stripe, we're actively working on VS Code as a first class citizen, but we know that it works with LSP plugins for emacs and vim, too.

**Comment by jez** [comment]
Closed in #1704
